addons/binding/org.openhab.binding.cm11a/src/main/java/org/openhab/binding/cm11a/handler/Cm11aApplianceHandler.java

```diff
@@ -0,0 +1,95 @@
+/**
+ * Copyright (c) 2014-2016 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.cm11a.handler;
+
+import java.io.IOException;
+
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.openhab.binding.cm11a.internal.InvalidAddressException;
+import org.openhab.binding.cm11a.internal.X10Interface;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Handler for Appliance (also called Switch) modules. These modules only support ON and OFF states
+ *
+ * @author Bob Raker
+ *
+ */
+public class Cm11aApplianceHandler extends Cm11aAbstractHandler {
+
+    private Logger logger = LoggerFactory.getLogger(Cm11aApplianceHandler.class);
+
+    private State desiredState = UnDefType.UNDEF;
+
+    public Cm11aApplianceHandler(Thing thing) {
+        super(thing);
+        this.thing = thing;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("**** Cm11aApplianceHandler handleCommand command = {}, channelUID = {}", command.toString(),
+                channelUID.getAsString());
+
+        x10Function = 0;
+        Bridge bridge = getBridge();
+        this.channelUID = channelUID;
+
+        Cm11aBridgeHandler cm11aHandler = (Cm11aBridgeHandler) bridge.getHandler();
+        if (cm11aHandler != null && cm11aHandler.getThing().getStatus().equals(ThingStatus.ONLINE)) {
+            if (command.equals(OnOffType.ON)) {
+                x10Function = X10Interface.FUNC_ON;
+                desiredState = OnOffType.ON;
+            } else if (command.equals(OnOffType.OFF)) {
+                x10Function = X10Interface.FUNC_OFF;
+                desiredState = OnOffType.OFF;
+            } else if (command instanceof RefreshType) {
+                x10Function = X10Interface.FUNC_OFF;
+                desiredState = OnOffType.OFF;
+                logger.info("Received REFRESH command for switch {}", houseUnitCode);
+            }
+
+            if (x10Function > 0) {
+                X10Interface x10Interface = cm11aHandler.getX10Interface();
+                x10Interface.scheduleHWUpdate(this);
+            } else {
+                logger.debug("Received invalid command for switch {} command: {}", houseUnitCode, command.toString());
```

Comment:
You don't have to call toString it will be called automatically, but only when the logging-level actually applies.

Please check all